File: apps/server/app/schemas/today.py
from datetime import date
from typing import Literal

from pydantic import BaseModel


class Greeting(BaseModel):
    name: str
    message: str
    subtitle: str


class TodayTask(BaseModel):
    id: int
    order: int
    title: str
    description: str
    category: str
    topic: str
    estimated_minutes: int


class UpcomingTask(BaseModel):
    id: int
    order: int
    title: str
    description: str
    estimated_minutes: int
    kind: Literal["build", "review"]


class StudySummary(BaseModel):
    total_minutes: int
    coding_minutes: int
    reading_minutes: int
    review_minutes: int


class Streak(BaseModel):
    days: int


class MasteryItem(BaseModel):
    name: str
    progress: int


class TodayResponse(BaseModel):
    date: date
    greeting: Greeting
    next_task: TodayTask
    upcoming_tasks: list[UpcomingTask]
    study_summary: StudySummary
    streak: Streak
    mastery: list[MasteryItem]

File: apps/desktop/src/types/today.ts
export type TodayTask = {
  id: number;
  order: number;
  title: string;
  description: string;
  category: string;
  topic: string;
  estimated_minutes: number;
};

export type UpcomingTask = {
  id: number;
  order: number;
  title: string;
  description: string;
  estimated_minutes: number;
  kind: "build" | "review";
};

export type MasteryItem = {
  name: string;
  progress: number;
};

export type TodayResponse = {
  date: string;
  greeting: { name: string; message: string; subtitle: string };
  next_task: TodayTask;
  upcoming_tasks: UpcomingTask[];
  study_summary: {
    total_minutes: number;
    coding_minutes: number;
    reading_minutes: number;
    review_minutes: number;
  };
  streak: { days: number };
  mastery: MasteryItem[];
};
